packages/block-editor/src/components/block-preview/index.js

```diff
@@ -9,38 +9,108 @@ import classnames from 'classnames';
  */
 import { Disabled } from '@wordpress/components';
 import { withSelect } from '@wordpress/data';
+import { useLayoutEffect, useState, useRef } from '@wordpress/element';
 
 /**
  * Internal dependencies
  */
 import BlockEditorProvider from '../provider';
 import BlockList from '../block-list';
+import { getBlockPreviewContainerDOMNode } from '../../utils/dom';
+
+export function BlockPreview( {
+	blocks,
+	settings,
+	scaleToFit = true,
+	scaleAdjustment = 1,
+	isScaled = true,
+} ) {
+	blocks = castArray( blocks );
+
+	const previewRef = useRef( null );
+
+	const [ isTallPreview, setIsTallPreview ] = useState( false );
+	const [ previewScale, setPreviewScale ] = useState( 1 );
+	const [ visibility, setVisibility ] = useState( 'hidden' );
+
+	// Dynamically calculate the scale factor
+	useLayoutEffect( () => {
+		if ( ! scaleToFit ) {
+			setVisibility( 'visible' );
+			return;
+		}
+		const blockClientIds = blocks.map( ( block ) => block.clientId );
+
+		// Timer - required to account for async render of `BlockEditorProvider`
+		const timerId = setTimeout( () => {
+			window.clearTimeout( timerId );
+			const previewContainer = previewRef.current;
+
+			if ( ! previewContainer ) {
+				return;
+			}
+
+			const comparisonBlock = blockClientIds.reduce( ( acc, currClientId ) => {
+				// Selector scoped to `previewContainerDomElement` to avoid global selector being ambiguous in the case
+				// of multiple previews on the same view
+				const previewDomElement = getBlockPreviewContainerDOMNode( currClientId, previewContainer );
```

Comment:
I guess this is done to pick the biggest block. I feel it's not needed. I think it could be as simple as saying, if we have more than one block (previewing templates), just pick the full width (700 right now)

Reply:
👍 Ok. in that case we need to find a dynamic way to define the width? We thought using a component property, let's say for instance `viewportWidth`, or something like that.